### PR TITLE
Add rectangle spline

### PR DESCRIPTION
**Pull request by beisaku**

### CODE REVIEW

File: ProceduralSplineComponent/Source/ProceduralSplineComponent/Public/RectangleSplineComponent.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Square Spline", meta = (ClampMin = 0.0f, EditCondition = "!bSquare"))
+	float Length = 500.0f;
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Square Spline", meta = (InlineEditConditionToggle))
+	bool bRoundCorner = false;
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Square Spline",
+		meta = (EditCondition = "bRoundCorner", DisplayName = "Rounded Corner"))
+	FSplineRoundedCornerParam RoundedCornerParam;
```
Comment (nutti) [inline ProceduralSplineComponent/Source/ProceduralSplineComponent/Public/RectangleSplineComponent.h]: Change `FSplineRoundedCornerParam` -> `FSplineRoundedCornerParams`

File: ProceduralSplineComponent/Source/ProceduralSplineComponent/Private/RectangleSplineComponent.cpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+		Length = Width;
+	}
+	InitRoundedCornerParam(RoundedCornerParam);
+
+	TArray<FVector> SquareVertexLocations;
+	SquareVertexLocations.SetNum(4);
+	SquareVertexLocations[0] = FVector(Width / 2.0f, Length / 2.0f, 0.0f);
+	SquareVertexLocations[1] = FVector(Width / 2.0f, -Length / 2.0f, 0.0f);
+	SquareVertexLocations[2] = FVector(-Width / 2.0f, -Length / 2.0f, 0.0f);
+	SquareVertexLocations[3] = FVector(-Width / 2.0f, Length / 2.0f, 0.0f);
```
Comment (nutti) [inline ProceduralSplineComponent/Source/ProceduralSplineComponent/Private/RectangleSplineComponent.cpp]: Consider to use `TStaticArray`

**Review by nutti — CHANGES_REQUESTED** [review]

File: ProceduralSplineComponent/Source/ProceduralSplineComponent/Private/RectangleSplineComponent.cpp (hunk 2 above)
Comment (beisaku) [reply ProceduralSplineComponent/Source/ProceduralSplineComponent/Private/RectangleSplineComponent.cpp]: Use Add

**Review by nutti — APPROVED** [review]
LGTM

**Pull request merged**